src/praisonai/praisonai/integrations/managed_local.py

```
@@ -468,14 +471,32 @@ def _install_packages(self) -> None:
-                logger.warning("[local_managed] pip install failed: %s", e.stderr)
-            except subprocess.TimeoutExpired:
-                logger.warning("[local_managed] pip install timed out")
+            
+            # Use compute provider if available, otherwise fall back to host
+            if self._compute:
+                try:
+                    # Route package installation through compute provider
+                    cmd_str = f"{sys.executable} -m pip install -q " + " ".join(pip_pkgs)
+                    result = self._compute.execute(cmd_str)
+                    logger.info("[local_managed] packages installed via compute provider")
+                except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)